pkg/kv/iterators_test.go

```diff
@@ -0,0 +1,307 @@
+package kv_test
+
+import (
+	"context"
+	"testing"
+
+	"github.com/go-test/deep"
+	"github.com/stretchr/testify/require"
+	"github.com/treeverse/lakefs/pkg/kv"
+	"github.com/treeverse/lakefs/pkg/kv/kvtest"
+)
+
+const (
+	firstPartitionKey  = "m"
+	secondPartitionKey = "ma"
+	thirdPartitionKey  = "mb"
+)
+
+func TestIterators(t *testing.T) {
+	ctx := context.Background()
+	store := kvtest.GetStore(ctx, t)
+	sm := kv.StoreMessage{
+		Store: store,
+	}
+	m := []kvtest.TestModel{
+		{Name: []byte("a")},
+		{Name: []byte("aa")},
+		{Name: []byte("b")},
+		{Name: []byte("c")},
+		{Name: []byte("d")},
+	}
+
+	// prepare data
+	for i := 0; i < len(m); i++ {
+		err := sm.SetMsg(ctx, firstPartitionKey, m[i].Name, &m[i])
+		if err != nil {
+			t.Fatal("failed to set model", err)
+		}
+
+		err = sm.SetMsg(ctx, secondPartitionKey, m[i].Name, &m[i])
+		if err != nil {
+			t.Fatal("failed to set model", err)
+		}
+	}
+
+	err := sm.SetMsg(ctx, thirdPartitionKey, []byte("e"), &kvtest.TestModel{Name: []byte("notin")})
+	if err != nil {
+		t.Fatal("failed to set model", err)
+	}
+
+	t.Run("partition iterator listing all values of partition", func(t *testing.T) {
+		itr := kv.NewPartitionIterator(ctx, store, (&kvtest.TestModel{}).ProtoReflect().Type(), firstPartitionKey)
+		require.NotNil(t, itr)
```

Comment:
Using 'testify/require' here why in prepare data we use 'if' - can we align in the test scope.